### PR TITLE
Add EnterpriseCatalog model

### PR DESCRIPTION
**Pull request by adamstankiewicz**
## Description

Adds `EnterpriseCatalog` model.

### CODE REVIEW

File: enterprise_catalog/apps/catalog/models.py [hunk 1]
```diff
@@ -41,4 +43,48 @@ def __str__(self):
...
+
+    .. no_pii:
+    """
+
+    uuid = models.UUIDField(
+        primary_key=True,
+        default=uuid4,
+        editable=False
+    )
+    enterprise_uuid = models.CharField(
```
Comment (bbaker6225) [inline enterprise_catalog/apps/catalog/models.py]: Are we linking with `uuid` or just `id`? If we're doing `uuid` I'd think we'd want to use a `UUIDField`, and if `id`then probably an `IntegerField`

File: enterprise_catalog/apps/catalog/models.py [hunk 2]
```diff
@@ -41,4 +43,48 @@ def __str__(self):
...
+        max_length=255,
+        blank=False,
+    )
+    content_catalog_query = models.ForeignKey(
+        CatalogQuery,
+        blank=False,
+        null=False,
```
Comment (bbaker6225) [inline enterprise_catalog/apps/catalog/models.py:65]: Could a `null` `CatalogQuery` potentially mean something like there is no restriction on the catalog? Not sure how that's currently represented as a query

File: enterprise_catalog/apps/catalog/models.py [hunk 3]
```diff
@@ -41,4 +43,48 @@ def __str__(self):
...
+        related_name='content_catalog_queries',
```
Comment (bbaker6225) [inline enterprise_catalog/apps/catalog/models.py]: I think we want the `related_name` to be `enterprise_catalog`.  I'd also suggest that we make the field name just `catalog_query` now that we changed the model name

File: enterprise_catalog/apps/catalog/models.py (hunk 2 above)
Comment (adamstankiewicz) [reply enterprise_catalog/apps/catalog/models.py:65]: A catalog with a content filter of an empty object (ie., `{}`) represents "All Content" or no restrictions on the catalog. If an enterprise has no restriction on that, there should be a shared catalog that represents "All Content"  for that case.

**Review by bbaker6225 — APPROVED** [review]

**Pull request merged**